form/unit_testing.rst

```diff
@@ -46,17 +46,21 @@ The simplest ``TypeTestCase`` implementation looks like the following::
                 'test' => 'test',
                 'test2' => 'test2',
             );
-
-            $form = $this->factory->create(TestedType::class);
-
+           
+           $objectToCompare = new TestObject();
+           //$objectToCompare will retrieve data from the form submission pass it at second argument
+            $form = $this->factory->create(TestedType::class, $objectToCompare);
+            
             $object = new TestObject();
             // ...populate $object properties with the data stored in $formData
 
             // submit the data to the form directly
             $form->submit($formData);
-
+            
+            //Will fill data in $objectToCompare
+            $objectToCompare = $form->getData();
```

Comment:
I would not introduce this variable here. There is no need for it.

Reply:
I understand but IMO it's strange to declare a variable, pass it as an argument and never use it after.

Reply:
But now we are doing exactly that because we are overwriting the previously declared variable.

Reply:
@xabbuh do we need to change anything in the merged PR? If yes, could you please create a quick issue (or PR) to not forget about it? Thanks a lot!

Reply:
see #9431